components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -146,10 +125,15 @@ export default function () {
                         </a>
                     </span>
                 }
-                checked={isIncrementalPrebuildsEnabled}
-                disabled={isLoading}
+                checked={!!projectSettings.useIncrementalPrebuilds}
                 onChange={toggleIncrementalPrebuilds}
             />
+            <CheckBox
+                title={<span>Cancel Prebuilds on Outdated Commits </span>}
+                desc={<span>Cancels all pending and running prebuilds on the branch when a new commit is pushed.</span>}
```

Comment:
In general, I think that this feature is super valuable and needed, and I'm happy that we're taking a first step towards it. 💯

Without having read the code much further or tested it, one initial thought that occurs is that, if one of your branches (e.g. `main`) takes a long time to build, and sees very frequent commits (e.g. during working hours), we might be always-cancelling running prebuilds without ever finishing one, for long stretches of time (e.g. maybe we only successfully finish running a prebuild over lunch break or at the end of the day).